src/lib/cache/subCache.cpp

```diff
@@ -1285,3 +1307,47 @@ void subCacheStart(void)
   }
   pthread_detach(tid);
 }
+
+
+
+extern bool noCache;
+/* ****************************************************************************
+*
+* subCacheItemNotificationErrorStatus - 
+*
+* This function marks a subscription to be erroneous, i.e. notifications are
+* not working.
+* A timestamp for this last failure is set for the sub-item in the sub-cache  and
+* the consecutive number of notification errors for the subscription is incremented.
+*
+* If 'errors' == 0, then the subscription is marked as non-erroneous.
+*/
+void subCacheItemNotificationErrorStatus(const std::string& tenant, const std::string& subscriptionId, int errors)
+{
+  if (noCache)
+  {
+    mongoSubCacheUpdate(tenant, subscriptionId, 0, 0, time(NULL), errors);
```

Comment:
The name of this function is not very fortunate.
What the function does is to update the fields in the database.
Let's find a better name ...

    mongoSubscriptionStatUpdate?

When implementing the lastSuccess/lastFailure, this function needs to be looked over.
I would propose NTC in this PR and to fix it in the next PR.